### DISCUSSION TITLE
IP remains accessible with additional action

### DISCUSSION THREAD

**Discussion by ipod86** [fail2ban/fail2ban#3828] [Q&A] [answered]
Hello, 

I have a problem with the following jail:

```
[planka]
enabled = true
port = http,https
filter = planka
logpath = /data/compose/46/logs/planka.log
maxretry = 1
bantime = 900
banaction = iptables-multiport
banaction_allports = iptables-allports
action = custom-iobroker-notify 
```

If I save like this, my action for iobroker is executed and IPs are banned if necessary. However, the IPs remain accessible. If I remove the action with iobroker, the IP is banned and is successfully banned and blocked. 

What could be the reason for this?

This is the action:

```
[Definition]
actionstart =
actionstop =
actioncheck =
actionban = curl -X GET "http://192.168.99.33:8087/set/0_userdata.0.Internet.fail2ban.fail2ban_action?value=<ip>:<bantime>:<failures>:<name>"
actionunban = curl -X GET "http://192.168.99.33:8087/set/0_userdata.0.Internet.fail2ban.fail2ban_action?value=<ip>:<name>"
 
```

**Comment by sebres** [comment, marked as answer]
Take an attentive look at many examples in jail.conf, for instance:
https://github.com/fail2ban/fail2ban/blob/1ea8a6de58e6d982b5e42db5b1761281e91c51e1/config/jail.conf#L236-L237
And to default `action` (and `action_`) declarations:
https://github.com/fail2ban/fail2ban/blob/1ea8a6de58e6d982b5e42db5b1761281e91c51e1/config/jail.conf#L212
https://github.com/fail2ban/fail2ban/blob/1ea8a6de58e6d982b5e42db5b1761281e91c51e1/config/jail.conf#L263

You have simply overwritten this default `action`, so it uses only your action and no banning action anymore.
You have to rewrite it:
```diff
- action = custom-iobroker-notify
+ action = %(action_)s
+          custom-iobroker-notify
```
